usr/Makefile

```diff
@@ -1,8 +1,42 @@
 # This Makefile will work only with GNU make.
 
+ifeq ($(TOPDIR),)
+	TOPDIR = ..
+endif
+
+OSNAME=$(shell uname -s)
+
+# allow users to override these
+# eg to compile for a kernel that you aren't currently running
+KERNELRELEASE ?= $(shell uname -r)
+KSRC ?= /lib/modules/$(KERNELRELEASE)/build
+
+KSUBLEVEL=$(shell cat $(KSRC)/Makefile | awk -F= '/^SUBLEVEL =/ {print $$2}' | \
+			sed 's/^[ \t]*//;s/[ \t]*$$//')
+
+ifeq ($(OSNAME),Linux)
+	ifeq ($(KSUBLEVEL),11)
+		IPC_CFLAGS=-DNETLINK_ISCSI=12 -D_GNU_SOURCE
+	else
+	ifeq ($(KSUBLEVEL),12)
+		IPC_CFLAGS=-DNETLINK_ISCSI=12 -D_GNU_SOURCE
+	else
+		IPC_CFLAGS=-DNETLINK_ISCSI=8 -D_GNU_SOURCE
+	endif
+	endif
+IPC_OBJ=netlink.o
+else
+ifeq ($(OSNAME),FreeBSD)
+IPC_CFLAGS=
+IPC_OBJ=ioctl.o
+endif
+endif
+
 CFLAGS ?= -O2 -g
 WARNFLAGS ?= -Wall -Wstrict-prototypes
-CFLAGS += $(WARNFLAGS) -I../include -I. -D_GNU_SOURCE
+CFLAGS += $(WARNFLAGS) -I../include -I. -D_GNU_SOURCE \
+	  -I$(TOPDIR)/libopeniscsiusr
+LDFLAGS += -L$(TOPDIR)/libopeniscsiusr -lopeniscsiusr
```

Comment:
Please do not set LDFLAGS directly, as many distros use this to set build-specific flags at run-time, which means that if anyone sets this on the make command line your library addition gets lost. Call it something like "OURLIB_LDFLAGS" (or whatever), then add that to the compile lines, e.g.:

>         $(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@  -lisns -lcrypto -lrt -lmount $(OURLIB_LDFLAGS)

Reply:
I don't get it. It still __appends__ to `LDFLAGS` even user defined it.